homeassistant/components/sonarr/config_flow.py

```diff
@@ -0,0 +1,144 @@
+"""Config flow for Sonarr."""
+import logging
+from typing import Any, Dict, Optional
+
+from sonarr import Sonarr, SonarrAccessRestricted, SonarrError
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    CONF_BASE_PATH,
+    CONF_UPCOMING_DAYS,
+    CONF_WANTED_MAX_ITEMS,
+    DEFAULT_BASE_PATH,
+    DEFAULT_PORT,
+    DEFAULT_SSL,
+    DEFAULT_UPCOMING_DAYS,
+    DEFAULT_VERIFY_SSL,
+    DEFAULT_WANTED_MAX_ITEMS,
+)
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+ERROR_CANNOT_CONNECT = "cannot_connect"
+ERROR_INVALID_AUTH = "invalid_auth"
+ERROR_UNKNOWN = "unknown"
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_API_KEY): str,
+        vol.Optional(CONF_BASE_PATH, default=DEFAULT_BASE_PATH): str,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
```

Comment:
Should this be limited to advanced flows?

Reply:
do you have example of how to do that?

Reply:
nvm found it https://developers.home-assistant.io/docs/data_entry_flow_index/#show-form